Mathlib/Combinatorics/SimpleGraph/CompleteMultipartite.lean

```diff
@@ -162,4 +169,121 @@ theorem IsCompleteMultipartite.comap {β : Type*} {H : SimpleGraph β} (f : H 
+    completeEquipartiteGraph r t ≃g completeMultipartiteGraph (Function.const (Fin r) (Fin t)) where
+  toFun := fun (v₁, v₂) ↦ by
+    use v₁, v₂, v₂.is_lt
+  invFun := fun ⟨v₁, v₂⟩ ↦ by
+    rw [Function.const_apply] at v₂
+    use v₁, v₂, v₂.is_lt
+  left_inv v := by simp
+  right_inv v := by simp
+  map_rel_iff' := by simp
+
+lemma completeEquipartiteGraph_adj {v w} :
+  (completeEquipartiteGraph r t).Adj v w ↔ v.1 ≠ w.1 := by rfl
+
+/-- `completeEquipartiteGraph r t` contains no edges when `r ≤ 1` or `t = 0`. -/
+lemma completeEquipartiteGraph_eq_bot_iff :
+    completeEquipartiteGraph r t = ⊥ ↔ r ≤ 1 ∨ t = 0 := by
+  rw [← not_iff_not, not_or, ← ne_eq, ← edgeSet_nonempty, not_le, ← Nat.succ_le_iff,
+    ← Fin.nontrivial_iff_two_le, ← ne_eq, ← Nat.pos_iff_ne_zero, Fin.pos_iff_nonempty]
+  refine ⟨fun ⟨e, he⟩ ↦ ?_, fun ⟨⟨i₁, i₂, hv⟩, ⟨x⟩⟩ ↦ ?_⟩
+  · induction' e with v₁ v₂
+    rw [mem_edgeSet, completeEquipartiteGraph_adj] at he
+    exact ⟨⟨v₁.1, v₂.1, he⟩, ⟨v₁.2⟩⟩
+  · use s((i₁, x), (i₂, x))
+    rw [mem_edgeSet, completeEquipartiteGraph_adj]
+    exact hv
+
+theorem completeEquipartiteGraph.isCompleteMultipartite :
+    (completeEquipartiteGraph r t).IsCompleteMultipartite := by
+  rcases t.eq_zero_or_pos with ht_eq0 | ht_pos
+  · rw [completeEquipartiteGraph_eq_bot_iff.mpr (Or.inr ht_eq0)]
+    exact bot_isCompleteMultipartite
+  · rw [isCompleteMultipartite_iff]
+    use (Fin r), Function.const (Fin r) (Fin t)
+    simp_rw [Function.const_apply, exists_prop]
+    exact ⟨Function.const (Fin r) (Fin.pos_iff_nonempty.mp ht_pos),
+      ⟨completeEquipartiteGraph.completeMultipartiteGraph⟩⟩
+
+theorem neighborSet_completeEquipartiteGraph (v) :
+    (completeEquipartiteGraph r t).neighborSet v = {v.1}ᶜ ×ˢ Set.univ := by
+  ext; simp [ne_comm]
+
+theorem neighborFinset_completeEquipartiteGraph (v) :
+    (completeEquipartiteGraph r t).neighborFinset v = {v.1}ᶜ ×ˢ univ := by
+  ext; simp [ne_comm]
+
+theorem degree_completeEquipartiteGraph (v) :
+    (completeEquipartiteGraph r t).degree v = (r-1) * t := by
+  rw [← card_neighborFinset_eq_degree, neighborFinset_completeEquipartiteGraph v,
+    card_product, card_compl, card_singleton, Fintype.card_fin, card_univ, Fintype.card_fin]
+
+theorem card_edgeFinset_completeEquipartiteGraph :
+    #(completeEquipartiteGraph r t).edgeFinset = (r.choose 2) * t^2 := by
+  rw [← mul_right_inj' two_ne_zero, ← sum_degrees_eq_twice_card_edges]
+  conv_lhs =>
+    rhs; intro v
+    rw [degree_completeEquipartiteGraph v]
+  rw [sum_const, smul_eq_mul, card_univ, card_prod, Fintype.card_fin, Fintype.card_fin]
+  conv_rhs =>
+    rw [← Nat.mul_assoc, Nat.choose_two_right, Nat.mul_div_cancel' r.even_mul_pred_self.two_dvd]
+  rw [← mul_assoc, mul_comm r _, mul_assoc t _ _, mul_comm t, mul_assoc _ t, ← pow_two]
+
+section Coloring
+
+/-- The injection `(x₁, x₂) ↦ x₁` is always a `r`-coloring of a `completeEquipartiteGraph r ·`. -/
+def Coloring.completeEquipartiteGraph :
```

Comment:
We currently have the colouring of a complete multipartite graph in Coloring.lean, it'd be nice to put this one there. But since this file imports Coloring, that wouldn't work; perhaps for now let's put this particular one in ConcreteColorings.lean, and the other one can be moved there later (since they should probably all be in there eventually)